e2e-tests/production-runtime/cypress/integration/redirects.js

```diff
@@ -75,7 +75,7 @@ describe(`Redirects`, () => {
       failOnStatusCode: false,
     }).waitForRouteChange()
 
-    cy.location(`pathname`).should(`equal`, `/redirect-search-hash/`)
+    cy.location(`pathname`).should(`equal`, `/redirect-search-hash`)
```

Comment:
Result of https://github.com/gatsbyjs/gatsby/pull/33209/files#r717559851

Please note that with this change it is actually same as we check client navigation in https://github.com/gatsbyjs/gatsby/blob/17a3f9f69ae2710f4aeef27603226281f2713281/e2e-tests/production-runtime/cypress/integration/redirects.js#L62-L71 

And same for 2 following changes